### PR TITLE
fix: openapi nullable references

### PR DESCRIPTION
**Pull request by Bram-dc**
A patch for [this issue](https://github.com/StefanTerdell/zod-to-json-schema/issues/103)

### CODE REVIEW

**Review by StefanTerdell — CHANGES_REQUESTED** [review]
Please add a test or two

File: src/parsers/nullable.ts [hunk 1]
```diff
@@ -47,6 +47,9 @@ export function parseNullableDef(
       currentPath: [...refs.currentPath],
     });
 
+    if (base && '$ref' in base)
+        return { allOf: [base], nullable: true } as any;
```
Comment (StefanTerdell) [inline src/parsers/nullable.ts:51]: Moving the result into allOf will affect its seen path, so need to account for this to not break references to this node

**Review by StefanTerdell — CHANGES_REQUESTED** [review]

File: src/parsers/nullable.ts (hunk 1 above)
Comment (StefanTerdell) [reply src/parsers/nullable.ts:51]: It should be safe to always use the allOf stuff so maybe we can assume that's where it will end up

Comment (Bram-dc) [reply src/parsers/nullable.ts:51]: > Moving the result into allOf will affect its seen path, so need to account for this to not break references to this node

I am not too experienced with json schemas. Is it allowed to reference a reference? And will it ever happend?

Comment (Bram-dc) [reply src/parsers/nullable.ts:51]: > In Draft 4-7, $ref behaves a little differently. When an object contains a $ref property, the object is considered a reference, not a schema. Therefore, any other properties you put in that object will not be treated as JSON Schema keywords and will be ignored by the validator. $ref can only be used where a schema is expected.
[https://json-schema.org/understanding-json-schema/structuring#dollarref](https://json-schema.org/understanding-json-schema/structuring#dollarref)

"$ref can only be used where a schema is expected."

**Comment by Bram-dc** [comment]
I am pretty sure all references are still valid. I have added 2 more tests for this possible double reference behaviour.

**Pull request merged**